avocado/utils/software_manager.py

```diff
@@ -150,7 +150,7 @@ def _init_on_demand(self):
                                'dnf': DnfBackend,
                                'zypper': ZypperBackend}
 
-            if backend_type not in backend_mapping.keys():
+            if backend_type not in list(backend_mapping.keys()):
```

Comment:
This shouldn't be needed in py2 nor py3. In what version this breaks?

Reply:
yes, but only where list is necessary. The `dict_keys` object implements `__in__` operator. Also the `keys` is also not needed at all as on all pythons the `in` operator on `dict` implies `is in keys`. So I'd just leave it as is or, in a separate commit, remove the `keys` completely.

Reply:
@ldoktor , ok, thanks for your explanation. I have updated.